app/models/mixins/supports_feature_mixin.rb

```diff
@@ -187,7 +187,12 @@ def define_supports_feature_methods(feature, is_supported: true, reason: nil, &b
           unsupported.delete(feature)
           if block_given?
             begin
-              instance_eval(&block)
+              result = instance_eval(&block)
+              # if no errors yet but result was false or an error message
+              # then add the error
+              if !unsupported.key?(feature) && (result.kind_of?(String) || result == false)
+                unsupported_reason_add(feature, SupportsFeatureMixin.reason_or_default(result))
+              end
```

Comment:
@chessbyte I was pretty psyched when the end solution was this simple. (along the way it was more complicated)